exporters/jaeger/src/recordable.cc

```diff
@@ -193,7 +193,33 @@ void JaegerRecordable::SetInstrumentationLibrary(
 void JaegerRecordable::AddLink(const trace::SpanContext &span_context,
                                const common::KeyValueIterable &attributes) noexcept
 {
-  // TODO: convert link to SpanRefernece
+  // Note: "The Link’s attributes cannot be represented in Jaeger explicitly."
+  // -- https://opentelemetry.io/docs/reference/specification/trace/sdk_exporters/jaeger/#links
+  //
+  // This implementation does not (currently) implement the optional conversion to span logs.
+
+  thrift::SpanRef reference;
+
+  reference.__set_refType(thrift::SpanRefType::FOLLOWS_FROM);
+
+  // IDs should be converted to big endian before transmission.
+  // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/jaeger.md#ids
+#if JAEGER_IS_LITTLE_ENDIAN == 1
+  reference.__set_traceIdHigh(
+      otel_bswap_64(*(reinterpret_cast<const int64_t *>(span_context.trace_id().Id().data()))));
+  reference.__set_traceIdLow(
+      otel_bswap_64(*(reinterpret_cast<const int64_t *>(span_context.trace_id().Id().data()) + 1)));
+  reference.__set_spanId(
+      otel_bswap_64(*(reinterpret_cast<const int64_t *>(span_context.span_id().Id().data()))));
+#else
+  reference.__set_traceIdLow(
+      *(reinterpret_cast<const int64_t *>(span_context.trace_id().Id().data())));
+  reference.__set_traceIdHigh(
+      *(reinterpret_cast<const int64_t *>(span_context.trace_id().Id().data()) + 1));
+  reference.__set_spanId(*(reinterpret_cast<const int64_t *>(span_context.span_id().Id().data())));
+#endif
+
+  references_.push_back(reference);
```

Comment:
how will this list of links get exported to the backend? Does some code need to be added in [ThriftSender::Append()](https://github.com/open-telemetry/opentelemetry-cpp/blob/c96a3e32f0b5bf17e58b1300383f9f205f33558f/exporters/jaeger/src/thrift_sender.cc#L23) to do that?

Reply:
Good catch! I missed a line when I pulled the changes over. Added: https://github.com/open-telemetry/opentelemetry-cpp/pull/1251/files#diff-3fc4e4d4cb58766986df79016e29275bf3712c2fdcda74c598152f42598bf546R43